From cec95ace7ebf8339940a500e07b4d1066624af39 Mon Sep 17 00:00:00 2001
From: "Erik M. Bray" <erik.bray@lri.fr>
Date: Tue, 11 Jun 2019 22:30:39 +0200
Subject: [PATCH] Trac #27970: Disable vpx support when building libgd for
 Cygwin

We don't use this functionality anyways.  If it happens to exist in the system
package that is also harmless.  It just happens that the system package for
libvpx is broken in some cases (Cygwin < 2.10).
---
 build/pkgs/libgd/spkg-install | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/build/pkgs/libgd/spkg-install b/build/pkgs/libgd/spkg-install
index 85cafb7f63..52271ce138 100644
--- a/build/pkgs/libgd/spkg-install
+++ b/build/pkgs/libgd/spkg-install
@@ -7,6 +7,14 @@ cd src
 
 export CFLAGS="-g $CFLAGS"
 
+if [ "$UNAME" = "CYGWIN" ]; then
+    # Compiling with vpx support creates a broken library in some cases
+    # because the vpx package itself is broken on some older Cygwin versions;
+    # we don't need this feature so safer to just disable
+    # https://trac.sagemath.org/ticket/27970
+    LIBGD_CONFIGURE="--without-vpx $LIBGD_CONFIGURE"
+fi
+
 # We explicitly disable X and fontconfig support, since (1) X is not a SAGE dependency,
 # and (2) the gd build fails on a lot of OS X PPC machines when X is enabled.
 sdh_configure --without-jpeg --without-xpm --without-x --without-fontconfig \
-- 
2.21.0

